lib/shortcode/configuration.rb

```diff
@@ -38,5 +41,22 @@ def initialize
     @self_closing_tags    = []
     @attribute_quote_type = '"'
     @use_attribute_quotes = true
+    @presenters = {}
+  end
+
+  def register_presenter(presenter)
+    validate_presenter presenter
+    [*presenter.for].each { |k| presenters[k.to_sym] = presenter }
   end
+
+  private
+
+    def validate_presenter(presenter)
+      raise ArgumentError, "The presenter must define the class method #for" unless presenter.respond_to?(:for)
+      raise ArgumentError, "The presenter must define an initialize method" unless presenter.private_instance_methods(false).include?(:initialize)
+      %w(content attributes).each do |method|
+        raise ArgumentError, "The presenter must define the method ##{method}" unless presenter.method_defined?(method.to_sym)
+      end
+    end
```

Comment:
I have moved the class methods from the Shortcode::Presenter class as instance methods of the Shortcode::Configuration class.